pyvista/themes.py

```diff
@@ -27,8 +41,35 @@ def set_plot_theme(theme):
         raise TypeError(f'Expected a pyvista.Theme or str, not '
                         f'a {type(theme)}')
 
+class _ThemeConfig():
+    """Provide common methods for theme configuration classes."""
+
+    @classmethod
+    def fromdict(cls, dict_):
+        """Create from a dictionary."""
+        inst = cls()
+        for key, value in dict_.items():
+            if not hasattr(inst, key):
+                raise KeyError(f'Invalid key "{key}" for {inst.__class__.__name__}')
+            attr = getattr(inst, key)
+            if hasattr(getattr(inst, key), 'fromdict'):
+                attr.fromdict(value)
```

Comment:
I still don't want to go into rigorous code review as long as this is a draft, but this seems wrong here: you instantiate a new `_ThemeConfig` subclass, then discard it. You probably need something like
```py
setattr(self, key, attr.fromdict(value))
```
(untested). The current implementation ignores changes to class-valued fields when doing something like
```py
theme = pv.themes.DefaultTheme()
theme.slider_style.classic.slider_length = 42
print(pv.themes.DefaultTheme.fromdict(theme._to_dict()) == theme)
# False
```
Assuming I understand correctly that this should be an idempotent roundtrip.